lib/core/Canvas.js

```diff
@@ -687,76 +536,105 @@ Canvas.prototype.toggleMarker = function(element, marker) {
 
+  if (!rootElement) {
+    rootElement = {
+      id: '__implicitroot_' + this._roots++,
+      children: [],
+      isImplicit: true
+    };
+  }
 
-// root element handling //////////////////////
+  this._ensureValid('root', rootElement);
 
-/**
- * Sets a given element as the new root element for the canvas
- * and returns the new root element.
- *
- * @param {Object|djs.model.Root} element
- * @param {boolean} [override] whether to override the current root element, if any
- *
- * @return {Object|djs.model.Root} new root element
- */
-Canvas.prototype.setRootElement = function(element, override) {
-  var activePlane = this._activePlane;
+  var layer = this.getLayer(rootElement.id, PLANE_LAYER_INDEX);
 
-  if (activePlane) {
-    return this.setRootElementForPlane(element, activePlane, override);
-  } else {
-    var basePlane = this.createPlane(BASE_LAYER, element);
+  svgAttr(layer, 'display', 'none');
 
-    this.setActivePlane(basePlane);
+  this._addRoot(rootElement, layer);
+
+  this._planes.push({
+    rootElement: rootElement,
+    layer: layer
+  });
 
-    return basePlane.rootElement;
+  return rootElement;
+};
+
+Canvas.prototype.removeRootElement = function(rootElement) {
+
+  var plane = this._findPlaneForRoot(rootElement);
+
+  if (!plane) {
+    return;
   }
+
+  // hook up life-cycle events
+  this._removeRoot(rootElement);
+
+  // cleanup layer
+  this._removeLayer(rootElement.id);
+
+  // clean up plane
+  this._planes = this._planes.filter(function(plane) {
+    return plane.rootElement !== rootElement;
+  });
 };
 
 
+// root element handling //////////////////////
+
 /**
  * Sets a given element as the new root element for the canvas
  * and returns the new root element.
  *
  * @param {Object|djs.model.Root} element
- * @param {Object|djs.model.Root} plane
  * @param {boolean} [override] whether to override the current root element, if any
  *
  * @return {Object|djs.model.Root} new root element
  */
-Canvas.prototype.setRootElementForPlane = function(element, plane, override) {
+Canvas.prototype.setRootElement = function(newRootElement, override) {
 
-  if (typeof plane === 'string') {
-    plane = this.getPlane(plane);
-  }
+  var currentRoot = this._rootElement;
 
-  if (element) {
-    this._ensureValid('root', element);
+  if (newRootElement === currentRoot) {
+    return;
   }
 
-  var currentRoot = plane.rootElement;
+  var plane;
 
-  if (currentRoot) {
-    if (!override) {
-      throw new Error('rootElement already set, need to specify override');
-    }
+  if (!newRootElement) {
+    throw new Error('newRootElement required');
+  }
+
+  plane = this._findPlaneForRoot(newRootElement);
 
-    this._removeRoot(currentRoot);
+  // give set add semantics for backwards compatibility
+  if (!plane) {
+    override = true;
   }
 
-  if (element) {
-    this._addRoot(element, plane);
+  if (override) {
+    newRootElement = this.addRootElement(newRootElement);
   }
```

Comment:
Why do we bind adding the element to `override`? If we support adding the element implicitly, I think we should make do it consistently when the element was not added before.

Reply:
The idea was to keep API compatibility with diagram-js < 7.4.

We can decide to ditch it all-together. Would be interesting to get your feedback.

Reply:
The use case is to "replace an existing root, in place".

Cf. bpmn-js Process -> collaboration.

Reply:
I would like to ditch it. Replace in place is still possible, but needs 2 canvas operations:
```
  canvas.setRootElement(newRoot);
  canvas.removeRootElement(oldRoot);
```

Using the API this way also makes it more robust against edge cases (e.g. active plane is a subprocess and the rootProcess is replaced with a collaboration via API)

Reply:
The removing of the root element still has not other side effects, so the adjustments in the `UpdateCanvasRootHandler` is straight forward

Reply:
Makes sense. I'll do that then. When is a better time to ditch then now, as we rework the API entirely?

Reply:
Yes, lets ditch it now. We have breaking changes anyway, so we can include it in diagram-js@8

Reply:
Ditched via https://github.com/bpmn-io/diagram-js/pull/601/commits/bd2149b951d9b79fcb406b3b20ed2690046bdad3.

Also, we do throw a descriptive error message on miss-use.